bigquery/main.go

```diff
@@ -83,16 +93,39 @@ func (bqf *actualBQFactory) Make(ctx context.Context) (bq, error) {
 	}
 	bqClient, err := bigquery.NewClient(ctx, projectID)
 	if err != nil {
-		return nil, fmt.Errorf("Error intiializing bigquery client: %v", err)
+		return nil, fmt.Errorf("Error intializing bigquery client: %v", err)
 	}
 	newClient := &actualBQ{client: bqClient}
 	return newClient, nil
 }
 
+func parseImageManifest(image string) (*buildImage, error) {
+	ref, err := name.ParseReference(image)
+	if err != nil {
+		return nil, fmt.Errorf("Error parsing image reference: %v", err)
+	}
+	img, err := remote.Image(ref, remote.WithAuthFromKeychain(google.Keychain))
+	if err != nil {
+		return nil, fmt.Errorf("Error obtaining image reference: %v", err)
+	}
+	sha, err := img.Digest()
+	layers, err := img.Layers()
+	totalSum := int64(0)
+	for _, layer := range layers {
+		layerSize, err := layer.Size()
+		if err != nil {
+			return nil, fmt.Errorf("Error parsing layer %v: %v", layer, err)
```

Comment:
Optimization: If this is making a bunch of request in serial, you may want to consider parallelizing it with goroutines and a WaitGroup or ErrGroup.

Reply:
We can ignore this optimization for now.